### ISSUE TITLE
Zoom Blackscreen and Won't Close

### ISSUE THREAD

**Issue by Zie0** [pop-os/cosmic-epoch#736]
The Zoom flatpak from the Cosmic app store will launch and prompt for a meeting password from a URL in a browser. After that point the meeting window appears as an empty black screen which cannot be closed with the GUI. I have to use the `kill` command to kill it or a system restart. I tried Zoom with the official Ubuntu flavored deb and had the same issue.

It appears this was mentioned in a similar issue with VirtualBox in https://github.com/pop-os/cosmic-epoch/issues/558 
That issue implies this may occur when automatic window tiling is on.

![screenshot-20240816_zoom_blackscreen](https://github.com/user-attachments/assets/7b1e01e0-19ae-4fbc-98ff-b16fbaf29d90)

```
% flatpak info us.zoom.Zoom                                                         :( 1 24-08-16 - 13:10:30

Zoom - Video Conferencing, Web Conferencing, Webinars, Screen Sharing

          ID: us.zoom.Zoom
         Ref: app/us.zoom.Zoom/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 6.1.6.1013
     License: LicenseRef-proprietary
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 10.9 MB
     Runtime: org.freedesktop.Platform/x86_64/23.08
         Sdk: org.freedesktop.Sdk/x86_64/23.08

      Commit: babd2544653d8b4cc150f8e856a6971e92b9d9d1e0e020093c88a431e267584b
      Parent: 1c74b3b1b17e512c6998c7cc1fde52161f9ebc5ac55be6a012f3b65e9bc41e8d
     Subject: Replace xdg-desktop-portal path in correct executable path (5e2ae78a)
        Date: 2024-08-15 18:58:32 +0000
```

**Comment by mkellyxp** [comment]
Whoops.. just reported the same issue!

**Comment by Zie0** [comment]
Hi @mkellyxp. It doesn't appear I can link this with the issue you just created, https://github.com/pop-os/cosmic-epoch/issues/739

**Comment by pabcrz** [comment]
> Whoops.. just reported the same issue!

Can you share screen as owner?

**Comment by alancleary** [comment]
Not sure whether or not this should be considered a separate issue; I have the same problem but not until I'm in a meeting, i.e. I can launch Zoom and login, etc, but once I'm in the meeting all windows are simply black. A few things worth noting:

* My computer is a System76 Adder WS
* I'm running Arch, not Pop!_OS
* This happens with both the current COSMIC release (epoch-1.0.0-alpha.1) and the git version available [on the AUR](https://aur.archlinux.org/packages/cosmic-session-git)
* This happens no matter how Zoom is installed: I've tried from the AUR, from Zoom's website, and using Snap
* I'm running the proprietary Nvidia driver
* Zoom works just fine when I switch to the GNOME desktop environment
* When a black window occurs, Zoom reports no errors in stderr or its log files

**Comment by Zie0** [comment]
> Not sure whether or not this should be considered a separate issue... I can launch Zoom and login, etc, but once I'm in the meeting all windows are simply black.

@alancleary I think this is the same issue. The only difference was I opened zoom from a meeting link and not from the "Join a meeting" screen. I just launched zoom first to replicate that I can also launch Zoom and login, but have the issue on  the meeting screen. I'm able to close the Zoom window and "Join a meeting" screen, but I cannot close the black meeting screen once I'm already in one without issuing the `kill` command to the process.

Thanks for your feedback. I updated the bug report so it's more clear to others whether or not they have this same issue.

**Comment by Zie0** [comment]
Tried again since I understood https://github.com/pop-os/cosmic-epoch/issues/801#issuecomment-2330860172 to mean the issue may have been resolved following an update on 3rd-4th September, however, I still had the issue after updating to have these package version

```
% apt list | grep cosmic   

budgie-wallpapers-cosmic/noble,noble 24.04 all
cosmic-app-library/noble,now 0.1.0~1724867842~24.04~db4a96e amd64 [installed,automatic]
cosmic-applets/noble,now 0.1.0~1725375368~24.04~0b53634 amd64 [installed,automatic]
cosmic-bg/noble,now 0.1.0~1725199523~24.04~876987a amd64 [installed,automatic]
cosmic-comp/noble,now 0.1~1725475842~24.04~79ae56c amd64 [installed,automatic]
cosmic-design-demo/noble 0.1.0~1704748391~24.04~d58cfad amd64
cosmic-edit/noble,now 0.1.0~1725374069~24.04~868dee2 amd64 [installed,automatic]
cosmic-files/noble,now 0.1.0~1725401857~24.04~b968bc0 amd64 [installed,automatic]
cosmic-greeter-daemon/noble,now 0.1.0~1724903468~24.04~cd9fc48 amd64 [installed,automatic]
cosmic-greeter/noble,now 0.1.0~1724903468~24.04~cd9fc48 amd64 [installed,automatic]
cosmic-icons/noble,now 0.1.0~1723836643~24.04~ea9e3b8 amd64 [installed,automatic]
cosmic-launcher/noble,now 0.1.0~1724339595~24.04~0a3e3f4 amd64 [installed,automatic]
cosmic-notifications/noble,now 0.1.0~1722531142~24.04~e9abef5 amd64 [installed,automatic]
cosmic-osd/noble,now 0.1.0~1721052322~24.04~27fc5e5 amd64 [installed,automatic]
cosmic-panel/noble,now 0.1.0~1723223917~24.04~e184d10 amd64 [installed,automatic]
cosmic-randr/noble,now 0.1.0~1722404448~24.04~71fabbb amd64 [installed,automatic]
cosmic-screenshot/noble,now 0.1.0~1721939965~24.04~031eb66 amd64 [installed,automatic]
cosmic-session/noble,now 0.1.0~1725300296~24.04~39c7f93 amd64 [installed,automatic]
cosmic-settings-daemon/noble,now 0.1.0~1725474506~24.04~1ed6880 amd64 [installed,automatic]
cosmic-settings/noble,now 0.1.0~1725454356~24.04~544da2b amd64 [installed,automatic]
cosmic-store/noble,now 0.1.0~1725457363~24.04~b13eed4 amd64 [installed,automatic]
cosmic-term/noble,now 0.1.0~1725375785~24.04~a71f0dc amd64 [installed]
cosmic-wallpapers/noble,noble,now 1.0.0~1719415313~24.04~0f2f16d all [installed,automatic]
cosmic-workspaces/noble,now 0.1.0~1723068086~24.04~70d6c41 amd64 [installed,automatic]
node-cosmiconfig/noble,noble 7.1.0+ds1-1 all
pop-de-cosmic/noble,now 3.0.0~1723139739~24.04~3393178 amd64 [installed,automatic]
ubuntu-mate-wallpapers-cosmic/noble,noble 24.04.0 all
ubuntu-wallpapers-cosmic/noble,noble 24.04.2 all
ubuntukylin-wallpapers-cosmic/noble,noble 24.04 all
ubuntustudio-wallpapers-cosmic/noble-updates,noble-updates 24.04.9 all
xdg-desktop-portal-cosmic/noble,now 0.1.0pop1~1724783238~24.04~e0458ec amd64 [installed,automatic]
```

# journalctl 
Including this in case it adds insight
```
ethan@pop-os:~
% journalctl -S "Sep 05 12:25:11"                                                                                                                                                                          24-09-05 - 12:25:39
Sep 05 12:25:11 pop-os brave-browser.desktop[12151]: Opening in existing browser session.
Sep 05 12:25:29 pop-os systemd[1748]: Started app-flatpak-us.zoom.Zoom-12564.scope.
Sep 05 12:25:30 pop-os systemd[1748]: Started app-flatpak-us.zoom.Zoom-12615.scope.

ethan@pop-os:~
% ps -efH | grep zoom | grep -v brave | awk '{print $2}' | xargs kill                                                                                                                                      24-09-05 - 12:25:55
kill: (13071): No such process

ethan@pop-os:~
% journalctl -S "Sep 05 12:25:11"                                                                                                                                                                   :( 123 24-09-05 - 12:26:04
Sep 05 12:25:11 pop-os brave-browser.desktop[12151]: Opening in existing browser session.
Sep 05 12:25:29 pop-os systemd[1748]: Started app-flatpak-us.zoom.Zoom-12564.scope.
Sep 05 12:25:30 pop-os systemd[1748]: Started app-flatpak-us.zoom.Zoom-12615.scope.
Sep 05 12:26:04 pop-os systemd[1748]: app-flatpak-us.zoom.Zoom-12564.scope: Consumed 3.087s CPU time.
```

**Comment by Zie0** [comment]
> > Whoops.. just reported the same issue!
> 
> Can you share screen as owner?

@alancleary I'm not able to share my screen using zoom. I haven't been able to get that far. My meeting windows appear blank so I don't have that as an option. I'm not entirely sure if that is the same issue as this one. Perhaps https://github.com/pop-os/cosmic-epoch/issues/802 may be a more appropriate place to report that?

**Comment by alancleary** [comment]
> @alancleary I'm not able to share my screen using zoom. I haven't been able to get that far. My meeting windows appear blank so I don't have that as an option. I'm not entirely sure if that is the same issue as this one. Perhaps #802 may be a more appropriate place to report that?

I can't get far enough to try and share my screen. I am aware of potential screen sharing issues on Linux that are likely unrelated to the issue we're experiencing here. I'll mention them in a comment on #802.

**Comment by alancleary** [comment]
@Zie0 I've fixed Zoom on my computer. Firstly, I set the experimental `nvidia_drm.fbdev=1` kernal parameter.<sup>[1](https://wiki.archlinux.org/title/NVIDIA#DRM_kernel_mode_setting)</sup> This didn't seem to make a difference but I'm noting it here for posterity. Second, I set a couple environment variables to force GBM to use the NVIDIA driver as its backend:<sup>[2](https://wiki.archlinux.org/title/Wayland#Requirements)</sup>
```
GBM_BACKEND=nvidia-drm
__GLX_VENDOR_LIBRARY_NAME=nvidia
```
This seems to be what actually fixed the problem. Here's a blog post that includes these steps and more when describing how to set up Wayland with the NVIDIA driver in (Arch) Linux: https://linuxiac.com/nvidia-with-wayland-on-arch-setup-guide/

As I noted before, Zoom works just fine for me in GNOME so I'm wondering if GNOME is detecting the graphics drivers and setting the GBM environment variables accordingly. If so, this seems like the type of thing that COSMIC could do as well. That said, as an Arch user I'm completely OK with setting the GBM backend myself.

**Comment by Zie0** [comment]
Thanks @alancleary -- with your help I worked around my issue too!

I'm using amd integrated graphics so I had to add to `__GLX_VENDOR_LIBRARY_NAME=mesa` to the end of my ` /etc/environment` file in order to get it to work

**Comment by alancleary** [comment]
> Thanks @alancleary -- with your help I worked around my issue too!
> 
> I'm using amd integrated graphics so I had to add to `__GLX_VENDOR_LIBRARY_NAME=mesa` to the end of my ` /etc/environment` file in order to get it to work

That's great! This seems like the type of thing that COSMIC aught to handle automatically since the UI supports switching between integrated and dedicated graphics on laptops, i.e. the manual intervention we both had to do isn't practical in the multi-gpu scenario.
